src/TestProblem/Hydro/CCSN/dtSolver_CCSN.cpp

```diff
@@ -50,43 +52,47 @@ double Mis_GetTimeStep_Lightbulb( const int lv, const double dTime_dt )
 
       const double dh = amr->dh[lv];
 
 #     pragma omp for schedule( runtime )
       for (int PID=0; PID<amr->NPatchComma[lv][1]; PID++)
       {
-         for (int k=0; k<PS1; k++)  {  const double z = amr->patch[0][lv][PID]->EdgeL[2] + (k+0.5)*dh;
-         for (int j=0; j<PS1; j++)  {  const double y = amr->patch[0][lv][PID]->EdgeL[1] + (j+0.5)*dh;
-         for (int i=0; i<PS1; i++)  {  const double x = amr->patch[0][lv][PID]->EdgeL[0] + (i+0.5)*dh;
-
-            const real Dens       = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[DENS][k][j][i];
-            const real Momx       = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[MOMX][k][j][i];
-            const real Momy       = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[MOMY][k][j][i];
-            const real Momz       = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[MOMZ][k][j][i];
-            const real Engy       = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[ENGY][k][j][i];
-#           ifdef DELE
-                  real dEint_Code = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[DELE][k][j][i];
-#           else
-                  real dEint_Code = NULL_REAL;
-#           endif
+         for (int k=0; k<PS1; k++)  {
+         for (int j=0; j<PS1; j++)  {
+         for (int i=0; i<PS1; i++)  {
+
+            const real Dens = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[DENS][k][j][i];
+            const real Momx = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[MOMX][k][j][i];
+            const real Momy = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[MOMY][k][j][i];
+            const real Momz = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[MOMZ][k][j][i];
+            const real Engy = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[ENGY][k][j][i];
 
 #           ifdef MHD
-            const real Emag = MHD_GetCellCenteredBEnergy( amr->patch[ amr->MagSg[lv] ][lv][PID]->magnetic[MAGX],
-                                                          amr->patch[ amr->MagSg[lv] ][lv][PID]->magnetic[MAGY],
-                                                          amr->patch[ amr->MagSg[lv] ][lv][PID]->magnetic[MAGZ],
-                                                          PS1, PS1, PS1, i, j, k );
+            const real Emag = MHD_GetCellCenteredBEnergyInPatch( lv, PID, i, j, k, amr->MagSg[lv] );
 #           else
             const real Emag = NULL_REAL;
 #           endif
 
             const real Eint_Code = Hydro_Con2Eint( Dens, Momx, Momy, Momz, Engy, true, MIN_EINT, Emag );
 
 
-//          call Src_Lightbulb() to get the heating/cooling rate if not computed yet
-            if ( dEint_Code == NULL_REAL )
+#           ifdef DEDT_LB
+            real dEint_Code = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[DEDT_LB][k][j][i];
+#           else
+            real dEint_Code = DEDT_UNINITIALIZED;
+#           endif
+
+
+//          call Src_Lightbulb() to compute the neutrino heating/cooling rate if not initialized yet
+            if ( dEint_Code == DEDT_UNINITIALIZED )
```

Comment:
I'm worried that amr->FluSg[lv] here may not be necessarily equal to that used during the code initialization (i.e., when setting DEDT_UNINITIALIZED in SetGridIC(). So how about checking both Sg=0 and 1 here?

If you agree, please also explain it by adding a comment.

Reply:
Agree. I add `dEint_Code_Old` which points to the data at `Sg = 1 - amr->FluSg[lv]` and check the values at both sandglasses.